roles/rsyslog/tasks/outputs/files.yml

```diff
@@ -1,8 +1,50 @@
 ---
+- name: Create __rsyslog_conf_files_output_rules 0
+  set_fact:
+    __rsyslog_conf_files_output_rules:
+      [ { "name": "output-files-default", "type": "output",
+          "sections": [ { "options": "{{ lookup('template', 'output_files_default.j2') }}" } ] } ]
+  loop: '{{ rsyslog_output_files }}'
+  when: item.path is not defined
+
+- name: Create __rsyslog_conf_files_output_rules 1
+  set_fact:
+    __rsyslog_conf_files_output_rules:
+      '{{ __rsyslog_conf_files_output_rules | d([]) }} +
+       [ { "name": "output-files-exc-{{ index }}", "type": "output",
+           "sections": [ { "options":
+             "ruleset(name=\"{{ item.name }}\") {\n
+               {{ item.facility | d("*") }}.{{ item.severity | d("*") }};{{ item.exclude }} {{ item.path }}\n}\n"
+             } ]
+       } ]'
+  loop: '{{ rsyslog_output_files }}'
```

Comment:
If you use the template module, probably you could get rid of `set_fact` altogether and use the map filter? https://jinja.palletsprojects.com/en/2.9.x/templates/#map